http-servers/src/main/java/com/palantir/remoting/http/server/WebPreconditions.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.http.server;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import javax.annotation.Nullable;
+import javax.ws.rs.BadRequestException;
+
+/**
+ * Wrapper around some of Guava's {@link Preconditions} to make its exceptions more http-friendly exceptions. It
+ * includes other useful web-oriented checks like {@link #checkNotEmpty(String)} and excludes less relevant ones (like
+ * {@link Preconditions#checkState(boolean)}.
+ */
+public final class WebPreconditions {
+
+    private WebPreconditions() {}
+
+    /**
+     * See {@link Preconditions#checkArgument(boolean, Object)}.
+     *
+     * @throws BadRequestException
+     *         when <code>expression</code> is false
+     */
+    public static void checkArgument(boolean expression, @Nullable Object message) {
+        try {
+            Preconditions.checkArgument(expression, message);
```

Comment:
I only went for this approach because I like Guava's dead simple and reliable templates and couldn't find a way of reusing them without the ugly try/catch. What do you think? Perhaps we should just remove invocations that use templates?

We could also rely on java.util.Formatter, but I worry about exposing such a complex class in exception handling code.